index.js

```diff
@@ -57,6 +73,34 @@ router
     ctx.status = 202
     ctx.body = 'OK'
   })
+  .get('/cloudflareError.png', async ctx => {
+    if (ctx.query['cloudflareErrorType'] == undefined || ctx.query['rayID'] == undefined || ctx.query['clientIP'] == undefined) {
```

Comment:
Better way to do this?  AFAICT checking for `undefined` in a dict is the fastest, if not the prettiest

Reply:
I _believe_ `key in object` is the fastest and accurately tests for the existence of the key, but doesn’t test the value.

```
~/C/A/reaction [master] » node
> const o = {foo:"bar", bar:undefined}
undefined
> if ("foo" in o) { console.log("ok") }
ok
undefined
> if ("bar" in o) { console.log("ok") }
ok
undefined
> if ("baz" in o) { console.log("ok") }
undefined
```